eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -712,6 +741,209 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
+          stay()
+        case (SpliceStatus.SpliceInProgress(txBuilder), msg: TxSignatures) =>
+          txBuilder ! InteractiveTxBuilder.ReceiveTxSigs(msg)
+          stay()
+        case (SpliceStatus.NoSplice, msg: TxSignatures) =>
+          d.commitments.latest.localFundingStatus match {
+            case dfu: LocalFundingStatus.DualFundedUnconfirmedFundingTx if dfu.sharedTx.txId == msg.txId => dfu.sharedTx match {
+              case fundingTx: PartiallySignedSharedTransaction => InteractiveTxBuilder.addRemoteSigs(dfu.fundingParams, fundingTx, msg) match {
+                case Left(cause) =>
+                  val unsignedFundingTx = fundingTx.tx.buildUnsignedTx()
+                  log.warning("received invalid tx_signatures for txid={} (current funding txid={}): {}", msg.txId, unsignedFundingTx.txid, cause.getMessage)
+                  // The funding transaction may still confirm (since our peer should be able to generate valid signatures),
+                  // so we cannot close the channel yet.
+                  stay() sending Error(d.channelId, InvalidFundingSignature(d.channelId, Some(unsignedFundingTx.txid)).getMessage)
+                case Right(fundingTx) =>
+                  log.info("publishing funding tx for channelId={} fundingTxId={}", d.channelId, fundingTx.signedTx.txid)
+                  val dfu1 = dfu.copy(sharedTx = fundingTx)
+                  val d1 = d.modify(_.commitments.active.at(0).localFundingStatus).setTo(dfu1)
+                  stay() using d1 storing() calling publishFundingTx(dfu1)
+              }
+              case _: FullySignedSharedTransaction =>
+                // Signatures are retransmitted on reconnection, but we may have already received them.
+                log.debug("ignoring duplicate tx_signatures for txid={}", msg.txId)
+                stay()
+            }
+            case _ =>
+              log.debug("ignoring unexpected tx_signatures for txid={}", msg.txId)
+              stay() sending Warning(d.channelId, UnexpectedInteractiveTxMessage(d.channelId, msg).getMessage)
+          }
+        case _ =>
+          log.info("ignoring unexpected interactive-tx message: {}", msg.getClass.getSimpleName)
+          stay() sending Warning(d.channelId, UnexpectedInteractiveTxMessage(d.channelId, msg).getMessage)
+      }
+
+    case Event(w: WatchFundingConfirmedTriggered, d: DATA_NORMAL) =>
+      acceptFundingTxConfirmed(w, d) match {
+        case Right((commitments1, _)) => stay() using d.copy(commitments = commitments1) storing() sending SpliceLocked(d.channelId, w.tx.txid)
+        case Left(_) => stay()
+      }
+
+    case Event(msg: SpliceLocked, d: DATA_NORMAL) =>
+      d.commitments.updateRemoteFundingStatus(msg.fundingTxid) match {
+        case Right((commitments1, _)) => stay() using d.copy(commitments = commitments1) storing()
+        case Left(_) => stay()
+      }
+
     case Event(INPUT_DISCONNECTED, d: DATA_NORMAL) =>
```

Comment:
Done in ff0eb797e.

Reply:
We also need to update our state to set `spliceStatus` to `SpliceStatus.NoSplice` + the error message you're currently using says "rbf attempt failed: disconnected" instead of "splice attempt failed: disconnected".

Reply:
Done in fd54c11d5.